components/places/src/api/places_api.rs

```diff
@@ -310,6 +318,57 @@ impl PlacesApi {
         Ok(sync_ping)
     }
 
+    // This is the new sync API until the sync manager lands. It's currently
+    // not wired up via the FFI - it's possible we'll do declined engines too
+    // before we do.
+    pub fn sync(
+        &self,
+        client_init: &sync15::Sync15StorageClientInit,
+        key_bundle: &sync15::KeyBundle,
+    ) -> Result<(ServiceStatus, telemetry::SyncTelemetryPing)> {
```

Comment:
Want to add the `SyncResult` struct from your RFC here, or hold off and keep the tuple for now?

I'm also not sure that a `Result<SyncResult>` (or `Result<ServiceStatus, ...>`) return type makes sense...what if we just returned a `SyncResult` directly, and then you could check `result.service_status == ServiceStatus::Ok` to see if it succeeded or not? (That would also let us return telemetry even in the failure case). It does make the function more difficult to work with; you can't use `?` directly. But maybe you can have a closure that uses `?`, unpack the result, and convert the error to a `ServiceStatus` once, here?

Reply:
Yes, that's a great idea and cleans things up significantly. The new version is much cleaner in terms of results etc, and paves a better path for not losing telemetry in the case of failure (although we do still drop it at the ffi layer for compat reasons)